### DISCUSSION TITLE
Is it possible to read discussions from public repositories?

### DISCUSSION THREAD

**Discussion by Rodien** [community/community#73142] [Discussions] [answered]
### Select Topic Area

Question

### Body

Hi,

I want to build a tool that looks at a public GitHub repository and retrieves all discussions and their properties. 

Would this be possible?

**Comment by ktauchathuranga** [comment]
i think yes,
using some webscraping magic
🖤

**Reply by Rodien** [reply]
Ah okay, so GitHub has currently no API I can use to accomplish this task?

**Comment by airtower-luna** [comment, marked as answer]
You can access discussions using the GraphQL API: https://docs.github.com/en/graphql/reference/objects#discussion

GraphQL takes some getting used to, but I can say from experience it's possible. :sweat_smile: Mind that GraphQL requests must be authenticated, even if you want to read data from public repos. See also: https://docs.github.com/en/graphql/guides/forming-calls-with-graphql

**Reply by Rodien** [reply]
Thank you! 

I did take a look at the docs before but I thought I would only be able to read discussions from my own repositories.
